### ISSUE TITLE
Create frontend structure

### ISSUE THREAD

**Issue by Malaguth** [Malaguth/skill-tracking#38]
# User Story

As a developer, I want to set up the frontend structure for a React project using TypeScript and organize it into page, feature, and component folders so that I can start building the application with a well-organized, scalable, and feature-oriented codebase.

## Description

This user story focuses on the initial setup of the frontend structure for a React project using TypeScript and a specific folder structure. The goal is to create a robust and efficient development environment that promotes feature-based development and clear separation of concerns. This includes setting up the basic directory structure with page, feature, and component folders, installing necessary dependencies, and configuring TypeScript for the project.

## Acceptance Criteria

1. A new repository should be created for the project.
2. The repository should be initialized with a package.json file by running `npm init`.
3. React and ReactDOM should be installed in the project as dependencies.
4. TypeScript should be installed as a devDependency.
5. A tsconfig.json file should be created and configured according to the project's needs.
6. A directory structure should be set up which includes at least:
   - components folder: inside src, for reusable UI elements
   - features folder: inside src, for each distinct functionality of the app
   - pages folder: inside src, for different views or screens of the app
7. Example React components for each type of folder (components, features, pages) should be created using TypeScript.
8. The application should run without errors when the command `npm start` is executed.
9. The project structure and setup, including the use of page, feature, and component folders, should be documented in the README file.

**Comment by Malaguth** [comment]
@jpbahiaz, could you help me define if using npm or another manager as yarn is better? Also, I think the current LTS version is the best option. Do you agree?

**Comment by jpbahiaz** [comment]
I agree. I'd suggest using pnpm as a package manager but I think we should stick with npm for simplicity.

**Comment by ViniciusLAAraujo** [comment]
## Technical Debt:
- Points discussed in the issue validation
### Internationalization
- Create a developer guide for the frontend. (update folder-structure.md)
- Update C4model with the new frontend structure.
- Replace Jest with vitest (look into the react testing library 1rst,how to use it with vitest).
- Remove "index" files and rename files to "comp", "styles", "types", "enums", and export default in each.
